homeassistant/components/tankerkoenig/sensor.py

```diff
@@ -125,9 +133,12 @@ def device_state_attributes(self):
             ATTR_HOUSE_NUMBER: self._house_number,
             ATTR_POSTCODE: self._postcode,
             ATTR_CITY: self._city,
-            ATTR_LATITUDE: self._latitude,
-            ATTR_LONGITUDE: self._longitude,
         }
+
+        if (self._show_on_map):
+            attrs[ATTR_LATITUDE] = self._latitude
+            attrs[ATTR_LONGITUDE] = self._longitude
```

Comment:
This does not only affect showing on map or not; it affects which attributes are shown by the entity or can be used by automations. E.g one automation could be coupled to a device tracker to send a notification with the prices for fuel stations close to that device's position.

Reply:
I checked different integrations like air visual, Luftdaten, nil, iss, entire and nmbs with show_on_map flag. I found this solution:
```
        if (self._show_on_map):
            attrs[ATTR_LATITUDE] = self._latitude
            attrs[ATTR_LONGITUDE] = self._longitude
        else:
            attrs["lat"] = self._latitude
            attrs["long"] = self._longitude
```
This works for me well. Do you agree?

Reply:
I'm not sure what's the best way to deal with this. After looking at the examples you mentioned, I saw that there is no consistency between integrations. Some don't add anything if "show_on_map" is false, some add "lat" and "long", another adds "lati" and "long",...
I'm ok with it either way; let's some maintainer have the last word.

Reply:
Ok thanks for taking a closer look. I prefer the initial approach because it's straight forward. Let's wait for the review.